src/main/org/deidentifier/arx/distributed/ARXPartition.java

```diff
@@ -69,164 +100,275 @@ public static Data getData(List<DataHandle> handles) {
+     *                       each including the original header.
      */
-    public static List<DataHandle> getPartitionsRandom(Data data, int number) {
-        
-        // Copy definition
-        DataDefinition definition = data.getDefinition();
-        
-        // Randomly partition
+    public static List<ARXPartition> getPartitionsRandom(Data data, int[] subsetIndices, int number) {
         DataHandle handle = data.getHandle();
-        Iterator<String[]> iter = handle.iterator();
-        String[] header = iter.next();
 
-        // Lists
-        List<List<String[]>> list = new ArrayList<>();
+        // Shuffle
+        List<String[]> markedRows = shuffleData(handle, subsetIndices);
+        return createPartitions(markedRows, handle.getDefinition(), number);
+    }
+
+    /**
+     * Splits a dataset into a specified number of partitions, ensuring that each partition
+     * includes the original header and is evenly distributed based on the total number of rows.
+     * The dataset is sorted before the splitting.
+     *
+     * @param data           The dataset to be partitioned.
+     * @param subsetIndices  An array of indices specifying a subset that is used during anonymization
+     * @param number         The number of partitions to create.
+     * @return               A list of {@link ARXPartition} objects representing the divided partitions,
+     *                       each including the original header.
+     */
+    public static List<ARXPartition> getPartitionsSorted(Data data, int[] subsetIndices, int number) {
+        DataHandle handle = data.getHandle();
+
+        // Collect indices to use for sorting
+        Set<String> qi = handle.getDefinition().getQuasiIdentifyingAttributes();
+        int[] qiIndices = createIndexArray(handle, qi);
+
+        // Sort
+        List<String[]> markedRows = sortData(handle, subsetIndices, qiIndices);
+        return createPartitions(markedRows, handle.getDefinition(), number);
+    }
+
+    /**
+     * Splits a list of marked rows into a specified number of partitions, creating ARXPartition objects.
+     * This method also handles the removal of markers from the rows and includes the header in each partition.
+     *
+     * @param markedRows      The list of data rows, each extended with a marker indicating whether it's part of the subset.
+     * @param definition      The DataDefinition object cloned from the original data, used for creating new Data objects.
+     * @param number          The number of partitions to create from the markedRows.
+     * @return                A list of {@link ARXPartition} objects representing the divided partitions.
+     *                        Each partition includes the original header and a subset of the data rows.
+     */
+    private static List<ARXPartition> createPartitions(List<String[]> markedRows, DataDefinition definition, int number) {
```

Comment:
Same comments as above. Changes make me nervous. I think that there are much simpler solutions.